AutoRest/Generators/Ruby/Azure.Ruby/TemplateModels/AzureMethodTemplateModel.cs

```diff
@@ -52,6 +55,89 @@ public bool IsLongRunningOperation
         }
 
         /// <summary>
+        /// Returns true if method has x-ms-pageable extension.
+        /// </summary>
+        public bool IsPageable
+        {
+            get { return Extensions.ContainsKey(AzureExtensions.PageableExtension); }
+        }
+
+        /// <summary>
+        /// Returns invocation string for next method async.
+        /// </summary>
+        public string InvokeNextMethodAsync()
+        {
+            RubyCodeNamer cn = new RubyCodeNamer();
+            var builder = new StringBuilder();
+            string nextMethodName;
+            var pageableExtension = JsonConvert.DeserializeObject<PageableExtension>(Extensions[AzureExtensions.PageableExtension].ToString());
+
+            Method nextMethod = null;
+            if (pageableExtension != null && !string.IsNullOrEmpty(pageableExtension.OperationName))
+            {
+                nextMethod = ServiceClient.Methods.FirstOrDefault(m =>
+                    pageableExtension.OperationName.Equals(m.SerializedName, StringComparison.OrdinalIgnoreCase));
+                nextMethodName = nextMethod.Name;
+            }
+            else 
+            {
+                nextMethodName = cn.GetMethodName((string)(Extensions["nextMethodName"]));
+                nextMethod = ServiceClient.Methods.Where(m => m.Name == nextMethodName).FirstOrDefault();
+            }
+
+            var origMethodGroupedParameters = Parameters.Where(p => p.Name.Contains(Name));
+            if (origMethodGroupedParameters.Count() > 0)
+            {
+                for (int j = 0; j < nextMethod.Parameters.Count; j++)
+                {
+                    Parameter param = nextMethod.Parameters[j];
+
+                    if (param.Name.Contains(nextMethod.Name) && (param.Name.Length > nextMethod.Name.Length)) //parameter that contains the method name + postfix, it's a grouped param
+                    {
+                        string argumentName = param.Name.Replace(nextMethodName, Name);
```

Comment:
should we explain bit in details, why are we changing name of the parameter on the fly...